var/spack/repos/builtin/packages/petsc/package.py

```diff
@@ -146,6 +148,7 @@ class Petsc(Package):
     depends_on('hdf5+mpi+hl+fortran', when='+hdf5+mpi')
     depends_on('zlib', when='+hdf5')
     depends_on('parmetis', when='+metis+mpi')
+    depends_on('valgrind', when='+valgrind+debug')
```

Comment:
So when `+valgrind~debug`, valgrind isn't needed?

Reply:
Hi,

Having +debug when using valgrind is advised but not needed. Removed.

Thanks